azafea/event_processors/endless/metrics/v3/model/_base.py

```diff
@@ -0,0 +1,523 @@
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+
+import logging
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from hashlib import sha512
+from typing import Any, Dict, Optional, Set, Tuple, Type, cast
+
+from gi.repository import GLib
+
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.declarative import declared_attr
+from sqlalchemy.ext.declarative.api import DeclarativeMeta
+from sqlalchemy.orm import relationship
+from sqlalchemy.orm.query import Query
+from sqlalchemy.schema import Column, ForeignKey
+from sqlalchemy.types import BigInteger, Boolean, DateTime, Integer, LargeBinary, Unicode
+
+from azafea.model import Base, DbSession
+
+from ..utils import get_child_values, get_event_datetime, get_variant
+
+
+log = logging.getLogger(__name__)
+
+VARIANT_TYPE = GLib.VariantType('(xxsa{ss}ya(aysxmv)a(aysyxxmv))')
+
+SINGULAR_EVENT_MODELS: Dict[str, Type['SingularEvent']] = {}
+AGGREGATE_EVENT_MODELS: Dict[str, Type['AggregateEvent']] = {}
+
+IGNORED_EVENTS: Set[str] = {
+    '005096c4-9444-48c6-844b-6cb693c15235',
+    '337fa66d-5163-46ae-ab20-dc605b5d7307',
+    '3a4eff55-5d01-48c8-a827-fca5732fd767',
+    '350ac4ff-3026-4c25-9e7e-e8103b4fd5d8',
+    '566adb36-7701-4067-a971-a398312c2874',
+    '5fae6179-e108-4962-83be-c909259c0584',
+    '6dad6c44-f52f-4bca-8b4c-dc203f175b97',
+    '72fea371-15d1-401d-8a40-c47f379f64fd',
+    '7be59566-2b23-408a-acf6-91490fc1df1c',
+    '8f70276e-3f78-45b2-99f8-94db231d42dd',
+    '91de63ea-c7b7-412c-93f3-6f3d9b2f864c',
+    '9a0cf836-12cd-4887-95d8-e48ccdf6e552',
+    '9c33a734-7ed8-4348-9e39-3c27f4dc2e62',
+    '9f06d0f7-677e-43ca-b732-ccbb40847a31',
+    'ab839fd2-a927-456c-8c18-f1136722666b',
+    'ae391c82-1937-4ae5-8539-8d1aceed037d',
+    'af3e89b2-8293-4703-809c-8e0231c128cb',
+    'b1f87a3f-a464-48d4-8e35-35dd45659010',
+    'b2b17dfd-c30e-4789-abcc-4a38323127f6',
+    'b89f9a4a-3035-4fc3-9bef-584367fe2c96',
+    'bef3d12c-df9b-43cd-a67c-31abc5361f03',
+    'c02a5764-7f81-48c7-aea4-1413fd4e829c',
+    'ce179909-dacb-4b7e-83a5-690480bf21eb',
+    'd936cd5c-08de-4d4e-8a87-8df1f4a33cba',
+    'e6541049-9462-4db5-96df-1977f3051578',
+    'fb59199e-5384-472e-af1e-00b7a419d5c2',
+}
+IGNORED_EMPTY_PAYLOAD_ERRORS: Set[str] = set()
+
+
+class Channel(Base):
+    __tablename__ = 'channel_v3'
+
+    id = Column(Integer, primary_key=True)
+
+    #: image ID (e.g. ``eos-eos3.1-amd64-amd64.170115-071322.base``)
+    image_id = Column(Unicode, nullable=False)
+    #: dictionary of string keys (such as ``facility``, ``city`` and
+    #: ``state``) to the values provided in the location.conf file (written by
+    #: the ``eos-label-location`` utility)
+    site = Column(JSONB, nullable=False)
+    #: dual boot computer
+    dual_boot = Column(Boolean, nullable=False)
+    #: live USB stick
+    live_usb = Column(Boolean, nullable=False)
```

Comment:
I would prefer this to just be "live". There are live sessions which are not on USB sticks, such as booting the ISO in a virtual machine.